### ISSUE TITLE
macOS: Xcode 9.3 build warnings

### ISSUE THREAD

**Issue by fanquake** [bitcoin/bitcoin#12867]
Building master (c564424d98b1561f16e633528172b3583719e5f3), these warnings are new as of macOS 10.13.4 and Xcode 9.3:
```
  OBJCXX   qt/bitcoin_qt-macdockiconhandler.o
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
  OBJCXX   qt/bitcoin_qt-macnotificationhandler.o
...
  CXXLD    bitcoind
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
  CXXLD    test/test_bitcoin
  CXXLD    bench/bench_bitcoin
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
  AR       qt/libbitcoinqt.a
  OBJCXXLD qt/bitcoin-qt
  CXXLD    qt/test/test_bitcoin-qt
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
```

**Comment by AkioNak** [comment]
The same warning was output in my environment (master ad960f577/macos 10.13.4/clang-902.0.39.1)
So, I tried to pass "-Wl, -pie" instead of "-pie" via $HARDENED_LDFLAGS,then the clang warning disappeared, but one new warning occurred in ld.
```
....
  CXXLD    libbitcoinconsensus.la
ld: warning: -pie being ignored. It is only used when linking a main executable
  AR       libbitcoin_server.a
...
```

**Comment by fanquake** [comment]
Fixed by #12899.
